frontend/src/utils/localesList.json

```diff
@@ -0,0 +1,10 @@
+[
+  {
```

Comment:
I think we need to generate the file, because before it reads on i18n.js, and Settings.jsx file

Reply:
If does not want to read the content of the folders at "real-time" (not at building) you need to generate the JSON file. Merge all translations in 1 file I think is not a good idea

Reply:
I add it before frontend build, but when u know say me smtg

Reply:
we use it

Reply:
We need this variable (I tried to remove It but It does not work)

Reply:
But I don't know why do you think it is unnecessary. There is only a possibility to add the JSON dynamically and is that... I think

Reply:
But the problem is not Crowdin, the problem is insert the lang without change code

Reply:
Ok, but forget Crowdin because if we want to use it, we need to use personalized scripts.

Yesterday I read about how do that, and the best solution is create the JSON file, to avoid process the folders to get the names each time you enter the website

Reply:
> * Leverage framework support: If React supports running code during compile time, similar to other frameworks, it could be the ideal solution. This way, you could automatically generate a language list from the translation folder, which can be used within the application.

CHAT-GPT response to: Does React support running code during compile time?

> No, React itself doesn't run code during compile time. However, tools commonly used with React, such as Webpack and Babel, do perform transformations and other operations during the "compile" phase (more accurately called the "bundling" or "transpilation" phase).
>
> When I mentioned "compile time" in the previous answer, I was referring to the phase when Webpack (or another bundler) processes your code to create a bundle. During this phase, you can use various plugins and loaders to manipulate your code, import assets, and more.
>
> For the specific task of dynamically importing translations based on the file structure, you'd typically use a combination of Node.js scripts (to generate some index or manifest file) and then use that in your React code. The React code itself doesn't execute at "compile time", but you can prepare certain things in advance (like generating an index of translations) to make it easier to import and use assets dynamically at runtime.

Reply:
But If you move it to build folder, how can you read it on react files?

Reply:
> Maybe this option could be a solution https://vitejs.dev/config/build-options.html#build-dynamicimportvarsoptions https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars

But how can you import a file that you generate at build time?